json_schemas/schema_validator.py

```diff
@@ -0,0 +1,121 @@
+import json
+from jsonschema import validate
+import os
+import traceback
+import argparse
+
+'''
+Grab the config and the schema to be tested if called from the command line.
+'''
+def parse_arguments():
+  parser = argparse.ArgumentParser(description='The Submitty config.json validator.')
+  parser.add_argument('--config', required=True, help='The path to the configuration to be validated')
+  parser.add_argument('--schema', required=True, help='The path to the schema to be validated against')
+  return parser.parse_args()
+
+
+'''
+Validates a config json piece by piece.
+'''
+def submitty_validate_schemas(config_json, schema):
+  submitty_validate_schema(config_json, schema['properties'], 'autograding', 'autograding', descriptive_title='global autograding block')
```

Comment:
You don't need to give a named argument here as it's the next one to be used anyway. I'd suggest removing the `descriptive_title=` bit here (and other places where its preceded by 4 arguments) and save the characters.